cmd/simulator/load/workers.go

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package load
+
+import (
+	"context"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"golang.org/x/sync/errgroup"
+)
+
+type WorkerGroup struct {
+	Workers []*Worker
+}
+
+func NewWorkerGroup(clients []ethclient.Client, senders []common.Address, txSequences [][]*types.Transaction) *WorkerGroup {
+	workers := make([]*Worker, len(clients))
+	for i, client := range clients {
+		workers[i] = NewWorker(client, senders[i], txSequences[i])
+	}
+
+	return &WorkerGroup{
+		Workers: workers,
+	}
+}
+
+func (wg *WorkerGroup) executeTask(ctx context.Context, f func(ctx context.Context, w *Worker) error) error {
+	eg := errgroup.Group{}
+	for _, worker := range wg.Workers {
+		worker := worker
+		eg.Go(func() error {
+			return f(ctx, worker)
+		})
+	}
+
+	return eg.Wait()
+}
+
+// IssueTxs concurrently issues transactions from each worker
+func (wg *WorkerGroup) IssueTxs(ctx context.Context) error {
+	return wg.executeTask(ctx, func(ctx context.Context, w *Worker) error {
+		return w.ExecuteTxsFromAddress(ctx)
+	})
+}
+
+// AwaitTxs concurrently waits for each worker to confirm the nonce of its last issued transaction.
+func (wg *WorkerGroup) AwaitTxs(ctx context.Context) error {
+	return wg.executeTask(ctx, func(ctx context.Context, w *Worker) error {
+		return w.AwaitTxs(ctx)
+	})
+}
+
+// ConfirmAllTransactions concurrently waits for each worker to confirm each transaction by checking
+// for it via eth_getTransactionByHash
+func (wg *WorkerGroup) ConfirmAllTransactions(ctx context.Context) error {
+	return wg.executeTask(ctx, func(ctx context.Context, w *Worker) error {
+		return w.ConfirmAllTransactions(ctx)
+	})
+}
+
+// Execute performs executes the following steps in order:
+// - IssueTxs
+// - AwaitTxs
+// - ConfirmAllTransactions
+func (wg *WorkerGroup) Execute(ctx context.Context) error {
+	start := time.Now()
+	defer func() {
+		log.Info("Completed execution", "totalTime", time.Since(start))
+	}()
+
+	executionStart := time.Now()
+	log.Info("Executing transactions", "startTime", executionStart)
+	if err := wg.IssueTxs(ctx); err != nil {
+		return err
+	}
+	awaitStart := time.Now()
```

Comment:
nit: could we add wspace before this or remove the one on line 86 for consistency?

Reply:
Agree with @darioush here. I'm a big fan of the empty line before and after `if`/`for`/`switch` blocks. Buuuut, not the end of the world. I'm also a big fan of following up a PR with logical changes with another PR address formatting nits...

Reply:
Updating this